src/github.com/getlantern/flashlight/genconfig/README.md

```diff
@@ -0,0 +1,83 @@
+# genconfig
+
+**Generate the YAML file and embedded Go datastructures containing the network configuration for the Lantern client**
+
+
+## How it works
+
+Config generation currently follows this approach:
+
+1. Build a *model*, a complete list of working chained servers, masquerades and proxied sites.
+
+1. Fill the Go code templates (*.go), based on the model.
+
+1. Fill the YAML template, based on the model. The YAML template already has filled in the LFS round-robin (flashlight servers for domain fronting).
+
+## Using genconfig
+
+### Setup
+
+You need the s3cmd tool installed and set up.  To install on Ubuntu:
+
+```bash
+sudo apt-get install s3cmd
+```
+
+On OS X:
+
+```bash
+brew install s3cmd
+```
+
+And then run `s3cmd --configure` and follow the on-screen instructions.  You
+can get AWS credentials that are good for uploading to S3 in
+[too-many-secrets/lantern_aws/aws_credential](https://github.com/getlantern/too-many-secrets/blob/master/lantern_aws/aws_credential).
+
+### Managing masquerade hosts
+
+The file domains.txt contains the list of masquerade hosts we use, and
+blacklist.txt contains a list of blacklisted domains that we exclude even if
+present in domains.txt.
+
+To alter the list of domains or blacklist:
+
+1. Edit [`domains.txt`](genconfig/domains.txt) and/or [`blacklist.txt`](genconfig/blacklist.txt)
+2. `go run genconfig.go -domains domains.txt -blacklist blacklist.txt`.
+3. Commit the changed [`masquerades.go`](config/masquerades.go) and [`cloud.yaml`](genconfig/cloud.yaml) to git if you want.
+4. *[Deprecated: we are not doing it any more]* Upload cloud.yaml to s3 using [`udpateyaml.bash`](genconfig/updateyaml.bash) if you want.
+
+### Managing proxied sites
+
+Lists of proxied sites are expected to live as text files in a directory, one
+domain per line.  You provide this directory to `genconfig` with the `-proxiedsites` argument.
+
+### Managing chained proxies
+
+The IPs, access tokens, and other details that clients need in order to
+connect to the chained (that is, non-fronted) proxies we run are contained in
+a JSON file that normally lives in `genconfig/fallbacks.json` and is fed to
+`genconfig` with the optional `-fallbacks` argument.
+
+You only to concern yourself with this when the list of chained proxies
+changes (e.g., when we launch or kill some server).  To learn how to reenerate
+the `fallbacks.json` file in that case, see [the relevant
+section](https://github.com/getlantern/lantern_aws#regenerating-flashlightgenconfigfallbackjson)
+of the README of the lantern_aws project.
+
+### Uploading to Redis
+
+To add a bunch of servers to the queue of a datacenter, so they'll get pulled by the config server as necessary,
```

Comment:
Adding servers to the queue is currently done automatically as servers are pulled from it, so we don't do this manually anymore.